### DISCUSSION TITLE
Multiple adjacent lines as a single paragraph in (enum) list

### DISCUSSION THREAD

**Discussion by Andrew15-5** [typst/typst#1594] [Q&A] [answered]
I have an issue with `enum()`. I want to make (an enumerated) list, but suppose lines are too long. I want to break a super long paragraph (enum item) into "<=80" lines, like I do in LaTeX. But I can't use `\`, or `\\` to tell Typst to not treat each line as a separate paragraph, but rather treat adjacent lines as a one paragraph.

```typst
// All doesn't work

+ line number one
line number two // (1)
+ line number three\
line number four // (2)

+ line number one\
line number two
+ line number three\\
line number four
```

However, looking at examples [here](https://typst.app/docs/reference/layout/enum), I tried to use multiple `[]` (for each item):

```typst
// Works

#enum[
  line number one
  line number two
][
  line number three
  line number four
] // (3)
```

But compared to (1) and (2) it looks much less readable and more LaTeX-like. Are there any other ways to define (enum) list with multiple adjacent lines as a single paragraph (which are clean looking)?

If list items are not single words (shopping list), then they are probably whole paragraphs worth of text. Therefore, I either have to use visual line wrapping in editor (`:set wrap` for Neovim) or (3). Since everyone can use any editor and Typst is "like any other programming languages", the rule of 80 could be applied by many document creators. But this rule breaks document layout ((1), (2)).

**Comment by laurmaedje** [comment, marked as answer]
Does this do what you want?
```
+ line number one \
  line number two 
+ line number three \
  line number four
```
In general, a list item's indent matters. All text that is indented as much as the first line will be part of the list item.

**Reply by Andrew15-5** [reply]
Almost. This is what I was looking for:

```typst
+ line number one
 line number two // 1 space also works
+ line number three
  line number four
```

The text represents lines in the source file, but they should be joined together (into enumerated paragraphs) in the output file.

I tried using more than 2 spaces, and all deep indented lines are used in a single paragraph. If we omit things like nested enum lists, then this is true:

>  All text that is indented as much **or deeper** as the first line will be part of the list item.

Does the 1 space indent counts as a bug or as a feature?

**Reply by laurmaedje** [reply]
It's a feature, but the plan is to go even further. Right now, the rule is that the indent must be at least as much as the end column of the list marker. The plan is to say that it only needs to be greater than the start column of the list marker. This way, you can consistently indent a list with 2 spaces even if your list marker is e.g. `12.` and thus 3 columns wide.
